.github/workflows/check-duplicates.yml

```diff
@@ -0,0 +1,26 @@
+name: Check duplicates
+
+# Run this workflow every time a new commit pushed to your repository
+
+on: [push, pull_request]
+
+jobs:
+  # Set the job key. The key is displayed as the job name
+  # when a job name is not provided
+  checkguard:
```

Comment:
This should have a different key (`checknames`?), and be part of `validate-header-guards.yml`. Both of the jobs are part of validating the header guards, so let's keep them together. (It also removes some shared boilerplate.)

Reply:
Refactored `validate-header-guards.yml` to avoid duplicated code.